block/src/Pos/DB/Block.hs

```diff
@@ -97,15 +97,40 @@ getTipBlock = getTipSomething "block" getBlock
 getSerializedBlock
     :: forall ctx m. (MonadRealDB ctx m)
     => HeaderHash -> m (Maybe ByteString)
-getSerializedBlock hh = do
-    DB{..} <- getBlockDataDB
-    liftIO $ Rocks.get rocksDB rocksReadOpts (blockKeyMod . sformat hashHexF $ hh)
+getSerializedBlock = getSerializedBlockOrUndo fst
 
 -- Get serialization of an undo data for block with given hash from Block DB.
 getSerializedUndo :: (MonadRealDB ctx m) => HeaderHash -> m (Maybe ByteString)
-getSerializedUndo hh = do
-    DB{..} <- getBlockDataDB
-    liftIO $ Rocks.get rocksDB rocksReadOpts (undoKeyMod . sformat hashHexF $ hh)
+getSerializedUndo = getSerializedBlockOrUndo snd
+
+getSerializedBlockOrUndo ::
+       MonadRealDB ctx m
+    => ((ByteString, ByteString) -> ByteString)
```

Comment:
Minor thing: I'm not a huge fan of this tuple and `snd` with `fst`, I think it would be better as a new type with two dedicated functions because it will be way harder to confuse what's block and what's undo.